src/Process/Pool/Server.php

```diff
@@ -26,9 +26,10 @@ public function start(): ProcessInterface
             $this->_getLogger()->info('Process pool server started.');
             $this->_getProcessPool()->start();
             while (true) {
-                $this->_getProcessSignal()->waitForSignal();
+                $this->_getProcessSignal()->processBufferedSignals();
+                sleep(1);
```

Comment:
Why one second? I imagine it's fine for the server process, but for the root won't it mean that our information about a child won't sync up for up to a second? Does that even matter?

Reply:
Just an arbitrary time to prevent the loop from spinning out of control. Signal will store all signals in a buffer asynchronously, so if a bunch of signals interrupt the process during that second + whatever time the loop takes in compute time the signals will be processed as a 'batch'.

I could make these `usleep`s or something if we think 1s is too long. Practically I saw not much of a difference locally and I would imagine that at scale it would be even less noticeable, probably when there is no work is when it would be most noticeable. There is a `process_pool_strategy.max_alarm_time` delay if there is absolutely nothing going on in the Server process world since it has to default to polling the db.

I'm open to suggestions, I have no data to go below a second yet though.